modules/siddhi-core/src/main/java/io/siddhi/core/query/input/stream/join/JoinProcessor.java

```diff
@@ -75,29 +92,43 @@ public void process(ComplexEventChunk complexEventChunk) {
                 } else if (eventType == ComplexEvent.Type.RESET) {
                     if (!leftJoinProcessor) {
                         StateEvent outputStateEvent = joinEventBuilder(null, streamEvent, eventType);
-                        returnEventChunkList.add(new ComplexEventChunk<>(
-                                outputStateEvent, outputStateEvent, true));
+                        returnEventChunkList.add(new JoinReturnEventChunk(new ComplexEventChunk<>(
+                                outputStateEvent, outputStateEvent, true), true));
                     } else {
                         StateEvent outputStateEvent = joinEventBuilder(streamEvent, null, eventType);
-                        returnEventChunkList.add(new ComplexEventChunk<>(
-                                outputStateEvent, outputStateEvent, true));
+                        returnEventChunkList.add(new JoinReturnEventChunk(new ComplexEventChunk<>(
+                                outputStateEvent, outputStateEvent, true), true));
                     }
                 } else {
                     joinStateEvent.setEvent(matchingStreamIndex, streamEvent);
-                    StreamEvent foundStreamEvent = findableProcessor.find(joinStateEvent, compiledCondition);
+
+                    StreamEvent foundStreamEvent;
+                    if (this.isOptimisedQuery) {
+                        try {
+                            foundStreamEvent = query(joinStateEvent);
+                        } catch (SiddhiAppRuntimeException e) {
+                            log.warn("Optimised join failed with '" + e.getMessage() + " in query '" + queryName +
```

Comment:
Shall we call the  Optimised join ... as Performing select clause in databases failed hence reverting back to querying only with where clause?

This is because users will not understand whats optimised mean

Reply:
+1. Fixed in 348a2be